app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java

```diff
@@ -546,27 +544,26 @@ public static WindowManager.LayoutParams retrievePopupLayoutParamsFromPrefs(
         popupLayoutParams.gravity = Gravity.LEFT | Gravity.TOP;
         popupLayoutParams.softInputMode = WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE;
 
-        final int centerX = (int) (player.getScreenWidth() / 2f - popupWidth / 2f);
-        final int centerY = (int) (player.getScreenHeight() / 2f - popupHeight / 2f);
+        final int centerX = (int) (playerUi.getScreenWidth() / 2f - popupWidth / 2f);
+        final int centerY = (int) (playerUi.getScreenHeight() / 2f - popupHeight / 2f);
         popupLayoutParams.x = popupRememberSizeAndPos
-                ? player.getPrefs().getInt(player.getContext().getString(
-                        R.string.popup_saved_x_key), centerX) : centerX;
+                ? prefs.getInt(context.getString(R.string.popup_saved_x_key), centerX) : centerX;
         popupLayoutParams.y = popupRememberSizeAndPos
-                ? player.getPrefs().getInt(player.getContext().getString(
-                        R.string.popup_saved_y_key), centerY) : centerY;
+                ? prefs.getInt(context.getString(R.string.popup_saved_y_key), centerY) : centerY;
 
         return popupLayoutParams;
     }
 
-    public static void savePopupPositionAndSizeToPrefs(final Player player) {
-        if (player.getPopupLayoutParams() != null) {
-            player.getPrefs().edit()
-                    .putFloat(player.getContext().getString(R.string.popup_saved_width_key),
-                            player.getPopupLayoutParams().width)
-                    .putInt(player.getContext().getString(R.string.popup_saved_x_key),
-                            player.getPopupLayoutParams().x)
-                    .putInt(player.getContext().getString(R.string.popup_saved_y_key),
-                            player.getPopupLayoutParams().y)
+    public static void savePopupPositionAndSizeToPrefs(final PopupPlayerUi playerUi) {
```

Comment:
Sounds like this and ``retrievePopupLayoutParamsFromPrefs`` should be inside the PopupPlayer or a PopupPlayerHelper.
I'm not sure if this is even accessed outside the Player, so I think it might be good if it's encapsulated/non-public.

Reply:
Moved to `PopupPlayerUi`